### PR TITLE
gh-121474: Add threading.Barrier parties arg sanity check.

### PR DESCRIPTION
**Pull request by pygeek**
* Issue: gh-121474

### CODE REVIEW

**Comment by pygeek** [comment]
> This matches the asyncio.Barrier now, which is great. But I notice neither handle 1.5 and other non-integral floats well

I noticed this too, but neither do any of the primitives in `threading`. My primary intention was to get Barrier in line with the other primitives.

I plan on creating a separate Discuss thread on the matter, once this current PR is merged. My guess is that since this behavior hasn't been a noted problem until now, that it wouldn't make sense to make sweeping changes to the primitives.

File: Lib/test/lock_tests.py [hunk 1]
```diff
@@ -1013,6 +1013,10 @@ def multipass(self, results, n):
         self.assertEqual(self.barrier.n_waiting, 0)
         self.assertFalse(self.barrier.broken)
 
+    def test_constructor(self):
+        self.assertRaises(ValueError, self.barriertype, parties = 0)
+        self.assertRaises(ValueError, self.barriertype, parties = -1)
```
Comment (ZeroIntensity) [inline Lib/test/lock_tests.py]:
```suggestion
        self.assertRaises(ValueError, self.barriertype, parties=-1)
```

File: Lib/test/lock_tests.py [hunk 2]
```diff
@@ -1013,6 +1013,10 @@ def multipass(self, results, n):
         self.assertEqual(self.barrier.n_waiting, 0)
         self.assertFalse(self.barrier.broken)
 
+    def test_constructor(self):
+        self.assertRaises(ValueError, self.barriertype, parties = 0)
```
Comment (ZeroIntensity) [inline Lib/test/lock_tests.py]:
```suggestion
        self.assertRaises(ValueError, self.barriertype, parties=0)
```

File: Lib/threading.py [hunk 3]
```diff
@@ -689,6 +689,8 @@ def __init__(self, parties, action=None, timeout=None):
         default for all subsequent 'wait()' calls.
 
         """
+        if parties < 1:
+            raise ValueError("parties must be > 0")
```
Comment (ZeroIntensity) [inline Lib/threading.py:693]:
```suggestion
            raise ValueError("parties must be greater than 0")
```

**Review by ZeroIntensity — CHANGES_REQUESTED** [review]
Looks good, other than a nitpick and PEP 8 compliance :slightly_smiling_face:

File: Lib/threading.py (hunk 3 above)
Comment (graingert) [reply Lib/threading.py:693]: Imho this should match the error from asyncio.Barrier

Comment (ZeroIntensity) [reply Lib/threading.py:693]: Yeah, that's a good idea, I didn't think about that. Ignore this suggestion then, @pygeek

**Comment by pygeek** [comment]
> Looks good, other than a nitpick and PEP 8 compliance 🙂

Nice catch. I don't think the linter runs on test files. I literally copied the check from https://github.com/python/cpython/pull/121480/files/58880e2d5387680385e3802d8e18d2d2de8ef470#diff-1d04f1b4512c48ecc14e3eae5713be5dea3bdbeba07f9d77bb08089103c880dbL754-L755 .

I'm sure the omission of the linter is intentional (speed).
Does it make sense to have a second lint check in CI that just checks Lib/tests?

**Comment by ZeroIntensity** [comment]
> Does it make sense to have a second lint check in CI that just checks Lib/tests?

Probably not, and out of the scope of this issue anyways.

**Comment by pygeek** [comment]
> > Does it make sense to have a second lint check in CI that just checks Lib/tests?
> 
> 
> 
> Probably not, and out of the scope of this issue anyways.

I was speaking more generally, outside the context of this PR. Anyway, I'll start a separate thread in Discuss on the topic.

**Review by ZeroIntensity — APPROVED** [review]
LGTM!

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @pygeek.

**Pull request merged**